### PR TITLE
DEPR: change DataFrame.append default sort kwarg

### PR DESCRIPTION
**Pull request by jbrockmendel**
xref #20613

AFAICT this doesn't break any tests, was this supposed to have been changed along with the default in `concat` #29786 ?

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -6678,14 +6678,13 @@ def append(self, other, ignore_index=False, verify_integrity=False, sort=None):
...
             Sort columns if the columns of `self` and `other` are not aligned.
-            The default sorting is deprecated and will change to not-sorting
-            in a future version of pandas. Explicitly pass ``sort=True`` to
-            silence the warning and sort. Explicitly pass ``sort=False`` to
-            silence the warning and not sort.
 
             .. versionadded:: 0.23.0
+            .. versionchanged:: 1.0.0
+
+           Changed to not sort by default.
```
Comment (jorisvandenbossche) [inline pandas/core/frame.py]:
```suggestion
            Changed to not sort by default.
```

**Comment by jreback** [comment]
> xref #20613
> 
> AFAICT this doesn't break any tests, was this supposed to have been changed along with the default in `concat` #29786 ?

IIRC this was changed *already* for 1.0, so *may* need to delay this

**Comment by jbrockmendel** [comment]
> IIRC this was changed already for 1.0, so may need to delay this

The last time the relevant part of the docstring was changed was in #20613, the same PR that deprecated the old pd.concat default behavior.

**Comment by jreback** [comment]
thanks @jbrockmendel must have been thinking about something else

**Pull request merged**